src/Hilke.KineticConvolution/Helpers/GeometryCalculator.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+
+
+namespace Hilke.KineticConvolution.Helpers
+{
+    public class GeometryCalculator<TAlgebraicNumber>
+    {
+        private readonly IAlgebraicNumberCalculator<TAlgebraicNumber> _calculator;
+        private readonly ConvolutionFactory<TAlgebraicNumber> _convolutionFactory;
+        private readonly VectorCalculator<TAlgebraicNumber> _vectorCalculator;
+
+        public GeometryCalculator(IAlgebraicNumberCalculator<TAlgebraicNumber> calculator)
+        {
+            _calculator = calculator ?? throw new ArgumentNullException(nameof(calculator));
+            _convolutionFactory = new ConvolutionFactory<TAlgebraicNumber>(_calculator);
+            _vectorCalculator = new VectorCalculator<TAlgebraicNumber>(_calculator);
+        }
+
+        public Tracing<TAlgebraicNumber>? CreateCornerArc(
+            Direction<TAlgebraicNumber> tangentDirectionBefore,
+            Point<TAlgebraicNumber> cornerPoint,
+            Direction<TAlgebraicNumber> tangentDirectionAfter,
+            TAlgebraicNumber radius)
+        {
+            var vc = _vectorCalculator;
+
+            var tangentBefore = vc.FromDirection(tangentDirectionBefore.Normalize());
+            var tangentAfter = vc.FromDirection(tangentDirectionAfter.Normalize());
+            var rightNormalBefore = vc.RotateThreeQuarterOfATurn(tangentBefore);
+            var rightNormalAfter = vc.RotateThreeQuarterOfATurn(tangentAfter);
+
+            var dotProduct = vc.GetDot(tangentBefore, rightNormalAfter);
+            if (_calculator.IsZero(dotProduct))
+            {
+                var haveSameDirections = _calculator.IsPositive(vc.GetDot(tangentBefore, tangentAfter));
+
+                if (!haveSameDirections && !_calculator.IsZero(radius))
+                {
+                    throw new ArgumentException(
+                        "For an extremity arc (360°), the radius must be zero.",
+                        nameof(radius));
+                }
+
+                return haveSameDirections ? null : createExtremityFullArc();
+            }
+
+            return createArcForNonAlignedPoints();
+
+            Arc<TAlgebraicNumber> createExtremityFullArc() =>
+                _convolutionFactory.CreateArc(
+                    center: cornerPoint,
+                    directions: _convolutionFactory.CreateDirectionRange(
+                        start: vc.ToDirection(rightNormalBefore),
+                        end: vc.ToDirection(rightNormalAfter),
+                                            orientation: Orientation.CounterClockwise),
+                    radius: _calculator.CreateConstant(0.0),
+                    weight: 1);
+
+            Arc<TAlgebraicNumber> createArcForNonAlignedPoints()
+            {
+                var isAcuteAngle = _calculator.IsPositive(dotProduct);
```

Comment:
It seems that an obtuse angle also yields a **true**.

Reply:
You're right. It actually was the sign of the angle. I updated the name to `IsAngleCounterClockwise`